.github/workflows/tests.yml

```
@@ -0,0 +1,155 @@
+        run: |
+          chmod +x gitstart
+          chmod +x tests/*.sh
+          chmod +x tests/*.bats
+      
+      - name: Run unit tests
+        run: |
+          bats tests/gitstart.bats
+      
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.10)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)